plugins/l4lb/l4lb.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2015 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"runtime"
+
+	"github.com/dcos/dcos-cni/pkg/spartan"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/containernetworking/cni/pkg/invoke"
+	"github.com/containernetworking/cni/pkg/ip"
+	"github.com/containernetworking/cni/pkg/ipam"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+)
+
+type NetConf struct {
+	types.NetConf
+	Args     map[string]interface{} `json:"args"`
+	MTU      int                    `json:"mtu"`
+	Delegate map[string]interface{} `json:"delegate"`
```

Comment:
Is there anything else that would indicate the type that Delegate might be? because then we can do a json RawMessage here and explicitly get the type later.

Who's defining the Delegate data structure? Also, what are the possible types, just string and map[string]string or are there more?